emscripten-simulated-jerry-api/jerry.js

```diff
@@ -161,29 +182,15 @@
 
     // Increase the reference count of the given jerry value
     acquire: function (jerry_value) {
-      this._getEntry(jerry_value).refCount++;
+      this._addRefCount(this._getEntry(jerry_value));
       return jerry_value;
     },
 
     // Decrease the reference count of the given jerry value and delete it if
     // there are no more internal references.
     release: function (ref) {
       var entry = this._getEntry(ref);
-      entry.refCount--;
-
-      if (entry.refCount <= 0) {
-        // Don't remove primitives like string, number, etc because
-        // we would otherwise loose the jerry_value that should be
-        // associated with it. (It's not possible to add the
-        // __jerryInternalProps property w/o 'boxing' the primitive).
-        // The value might get 'sent back' to the native side later on
-        // and we want to make sure the jerry_value remains the same for
-        // the entirety of the lifetime of the value.
-        if (hasInternalProps(entry.value)) {
-          // console.log('deleting ' + ref + ' at ' + stackTrace());
-          delete this._jerryToHostValueMap[ref];
```

Comment:
I think we can the entry in the `_jerryhandleGarbageCollected` callback?

Reply:
yes, will do that